mne/io/reference.py

```diff
@@ -35,23 +38,17 @@ def _copy_channel(inst, ch_name, new_ch_name):
     inst : instance of Raw | Epochs | Evoked
         The data with a copy of a given channel.
     """
-    idx_ch = inst.ch_names.index(ch_name)
-    data = inst._data
-    data_ch = data[..., idx_ch, :].reshape((1, data.shape[-1]))
-    new_ch_info = create_info([new_ch_name], inst.info['sfreq'], ['eeg'])
-    if issubclass(type(inst), BaseRaw):
-        new_ch = RawArray(data_ch, new_ch_info, verbose=0)
-    elif issubclass(type(inst), BaseEpochs):
-        new_ch = EpochsArray(data_ch, new_ch_info, verbose=0)
-    else:
-        new_ch = EvokedArray(data_ch, new_ch_info, verbose=0)
-    inst.add_channels([new_ch], force_update_info=True)
+    new_inst = inst.copy().pick_channels([ch_name])
```

Comment:
This was my first idea but since data is preloaded, I wanted to avoid using `inst.copy()`. It seemed more "memory-efficient" at the time. This way is probably "nicer", though.